flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/batch/connectors/hive/HiveTableOutputFormat.java

```diff
@@ -190,7 +191,8 @@ public void finalizeGlobal(int parallelism) throws IOException {
 		StorageDescriptor jobSD = hiveTablePartition.getStorageDescriptor();
 		Path stagingDir = new Path(jobSD.getLocation());
 		FileSystem fs = stagingDir.getFileSystem(jobConf);
-		try (HiveMetastoreClientWrapper client = HiveMetastoreClientFactory.create(new HiveConf(jobConf, HiveConf.class))) {
+		String hiveVersion = jobConf.get(HiveCatalogValidator.CATALOG_HIVE_VERSION, null);
```

Comment:
can this be a member variable that serves in lifetime of the output format?

And we can reuse it in `moveFiles()`